File: src/g_loc/java/entities/Reservation.java
package entities;

import java.time.LocalDate;
import java.time.LocalTime;

public class Reservation {

    private int id;
    private double prix;
    private int local_id;  // Use camelCase for variable names
    private int utilisateur_id;  // Use camelCase for variable names
    private LocalDate date;
    private LocalTime heureDebut;
    private LocalTime heureFin;
    private boolean isConfirmee;  // Using 'isConfirmed' instead of 'isConfirmee'

    // Constructor
    public Reservation( int id, int local_id, int utilisateur_id, LocalDate date, LocalTime heureDebut, LocalTime heureFin, boolean isConfirmee) {
        this.id = id;
        this.local_id = local_id;
        this.utilisateur_id = utilisateur_id;
        this.date = date;
        this.heureDebut = heureDebut;
        this.heureFin = heureFin;
        this.isConfirmee = isConfirmee;
    }
    // Add this constructor to Reservation.java

    // Constructor with price
    public Reservation(int id, int local_id, int utilisateur_id, LocalDate date, LocalTime heureDebut, LocalTime heureFin, boolean isConfirmee, double prix) {
        this.id = id;
        this.local_id = local_id;
        this.utilisateur_id = utilisateur_id;
        this.date = date;
        this.heureDebut = heureDebut;
        this.heureFin = heureFin;
        this.isConfirmee = isConfirmee;
        this.prix = prix;
    }

    public Reservation(int local, String date, double prix) {
        this.local_id = local;
        this.date = LocalDate.parse(date);
        this.prix = prix;
    }


    // Getters and setters
    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }
public void setConfirmee(boolean confirmee) {this.isConfirmee = confirmee;}
    public int getLocalId() {
        return this.local_id;
    }
    public double getPrix() {
        return prix;
    }
    public void setPrix(double prix) {
        this.prix = prix;
    }
    public void setLocalId(int localId) {
        this.local_id = localId;
    }

    public int getUtilisateurId() {
        return utilisateur_id;
    }

    public void setUtilisateurId(int utilisateurId) {
        this.utilisateur_id = utilisateurId;
    }

    public LocalDate getDate() {
        return date;
    }

    public void setDate(LocalDate date) {
        this.date = date;
    }

    public LocalTime getHeureDebut() {
        return heureDebut;
    }

    public void setHeureDebut(LocalTime heureDebut) {
        this.heureDebut = heureDebut;
    }

    public LocalTime getHeureFin() {
        return heureFin;
    }

    public void setHeureFin(LocalTime heureFin) {
        this.heureFin = heureFin;
    }

    public boolean isConfirmee() {
        return isConfirmee;
    }

    public void setConfirmed(boolean isConfirmee) {
        this.isConfirmee = isConfirmee;
    }

}

File: src/Entity/Reservation.php
<?php

namespace App\Entity;

use App\Repository\ReservationRepository;
use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity(repositoryClass: ReservationRepository::class)]
#[ORM\Table(name: 'reservation')]
class Reservation
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[ORM\Column(type: 'date')]
    private ?\DateTimeInterface $date = null;

    #[ORM\Column(name: 'heureDebut', type: 'time')]
    private ?\DateTimeInterface $heureDebut = null;

    #[ORM\Column(name: 'heureFin', type: 'time')]
    private ?\DateTimeInterface $heureFin = null;

    #[ORM\Column(type: 'decimal', precision: 10, scale: 2)]
    private ?string $prix = null;

    #[ORM\Column(name: 'isConfirmee', type: 'boolean')]
    private ?bool $isConfirmee = null;

    #[ORM\Column(length: 50)]
    private ?string $statut = null;

    #[ORM\Column(name: 'created_at', type: 'datetime')]
    private ?\DateTimeInterface $createdAt = null;

    // Relations
    #[ORM\ManyToOne(targetEntity: Local::class)]
    #[ORM\JoinColumn(name: 'local_id', referencedColumnName: 'id', nullable: false)]
    private ?Local $local = null;

    #[ORM\ManyToOne(targetEntity: Utilisateur::class)]
    #[ORM\JoinColumn(name: 'utilisateur_id', referencedColumnName: 'id', nullable: false)]
    private ?Utilisateur $utilisateur = null;

    // ------------------ Getters & Setters ------------------

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getDate(): ?\DateTimeInterface
    {
        return $this->date;
    }

    public function setDate(\DateTimeInterface $date): static
    {
        $this->date = $date;
        return $this;
    }

    public function getHeureDebut(): ?\DateTimeInterface
    {
        return $this->heureDebut;
    }

    public function setHeureDebut(\DateTimeInterface $heureDebut): static
    {
        $this->heureDebut = $heureDebut;
        return $this;
    }

    public function getHeureFin(): ?\DateTimeInterface
    {
        return $this->heureFin;
    }

    public function setHeureFin(\DateTimeInterface $heureFin): static
    {
        $this->heureFin = $heureFin;
        return $this;
    }

    public function getPrix(): ?string
    {
        return $this->prix;
    }

    public function setPrix(string $prix): static
    {
        $this->prix = $prix;
        return $this;
    }

    public function isIsConfirmee(): ?bool
    {
        return $this->isConfirmee;
    }

    public function setIsConfirmee(bool $isConfirmee): static
    {
        $this->isConfirmee = $isConfirmee;
        return $this;
    }

    public function getStatut(): ?string
    {
        return $this->statut;
    }

    public function setStatut(string $statut): static
    {
        $this->statut = $statut;
        return $this;
    }

    public function getCreatedAt(): ?\DateTimeInterface
    {
        return $this->createdAt;
    }

    public function setCreatedAt(\DateTimeInterface $createdAt): static
    {
        $this->createdAt = $createdAt;
        return $this;
    }

    public function getLocal(): ?Local
    {
        return $this->local;
    }

    public function setLocal(Local $local): static
    {
        $this->local = $local;
        return $this;
    }

    public function getUtilisateur(): ?Utilisateur
    {
        return $this->utilisateur;
    }

    public function setUtilisateur(Utilisateur $utilisateur): static
    {
        $this->utilisateur = $utilisateur;
        return $this;
    }
}
